content/app/github/window-helper.js

```diff
@@ -1,51 +1,129 @@
+	var Traversing = require("content/app/algo/three/traversing");
+	var FindTextRangesVisitor = require("content/app/algo/three/fint-text-ranges-visitor");
 
 	/**
 	 * Helper for accessing and controlling window and root page elements.
 	 */
-	var WindowHelper = Class(Object, {
+	var WindowHelper = Class(EventDispatcher, {
 
 		$statics : {
 
-			/**
-			 * Get viewport bounding box.
-			 * @return {Box}
-			 */
-			getViewportBoundingBox : function () {
-				var width = document.documentElement.clientWidth;
-				var height = document.documentElement.clientHeight;
-				var left = document.body.scrollLeft;
-				var top = document.body.scrollTop;
-				return new Box(left, top, width, height);
-			},
-
-			/**
-			 * Get viewport bounding box relative to client visible area.
-			 * That means that viewport is at position (0,0).
-			 * @return {Box}
-			 */
-			getViewportClientBoundingBox : function () {
-				var width = document.documentElement.clientWidth;
-				var height = document.documentElement.clientHeight;
-				return new Box(0, 0, width, height);
-			},
-
-			/**
-			 * Scrolls vertically document body.
-			 * @param  {Number}		scrollDelta
-			 */
-			scrollTop : function(value) {
-				document.body.scrollTop = value;
-			},
+			INTERVAL_TICK : 100,	// ms
 
+		},
+
+		/**
+		 * Helper constructor.
+		 */
+		constructor : function () {
+			WindowHelper.$super.call(this);
+
+			this._selectedText = '';
+			this._intervalRef = setInterval(
+				this._onIntervalTick.bind(this),
+				WindowHelper.INTERVAL_TICK
+			);
+		},
+
+		/**
+		 * Get viewport bounding box.
+		 * @return {Box}
+		 */
+		getViewportBoundingBox : function () {
+			var width = document.documentElement.clientWidth;
+			var height = document.documentElement.clientHeight;
+			var left = document.body.scrollLeft;
+			var top = document.body.scrollTop;
+			return new Box(left, top, width, height);
+		},
+
+		/**
+		 * Get viewport bounding box relative to client visible area.
+		 * That means that viewport is at position (0,0).
+		 * @return {Box}
+		 */
+		getViewportClientBoundingBox : function () {
+			var width = document.documentElement.clientWidth;
+			var height = document.documentElement.clientHeight;
+			return new Box(0, 0, width, height);
+		},
+
+		/**
+		 * Scrolls vertically document body.
+		 * @param  {Number}		scrollDelta
+		 */
+		scrollTop : function(value) {
+			document.body.scrollTop = value;
+		},
+
+		/**
+		 * Append DOM element to page.
+		 * @param  {Element} child
+		 */
+		appendChild : function (child) {
+			document.body.appendChild(child);
+		},
+
+		/**
+		 * Remove DOM element from page.
+		 * @param  {Element} child
+		 */
+		removeChild : function (child) {
+			document.body.removeChild(child);
+		},
+
+		/**
+		 * Find ranges of provided text in page.
+		 * @param  {String} 	text
+		 * @return {Range[]}
+		 */
+		findTextRanges : function (text) {
+			return WindowHelper.findTextRangesInElement(document.body, text);
```

Comment:
Are you absolutely sure that we need to traverse document.body? Remember that this page contains multiple tabs! I guess this should be part of CommitPageHelper?